### PR TITLE
Restore status text and screen reader output

### PR DESCRIPTION
**Pull request by shoogle**
Resolves:

- https://musescore.org/en/node/306633
- https://musescore.org/en/node/306726

Fixes bugs and regressions from PR #6173.

- [x] I signed [CLA](https://musescore.org/en/cla)
- [x] I made sure the code in the PR follows [the coding rules](https://github.com/musescore/Documentation/blob/master/WorkflowAndGuidelines/CodeGuidelines.md)
- [x] I made sure the code compiles on my machine
- [x] I made sure there are no unnecessary changes in the code
- [x] I made sure the title of the PR reflects the core meaning of the issue you are solving
- [x] I made sure the commit message(s) contain a description and answer the question "Why do those changes fix that particular issue?" or "Why are those changes really necessary as improvements?"
- [x] I made sure the commit message title starts with "fix #424242:" if there is a related issue
- [N/A] I created the test (mtest, vtest, script test) to verify the changes I made

### CODE REVIEW

File: mscore/scoreaccessibility.cpp [hunk 1]
```diff
@@ -352,8 +352,7 @@ void ScoreAccessibility::updateAccessibilityInfo()
       if (!w)
             return;
 
-      if (w->score()->accessibleMessage().isEmpty())
-            currentInfoChanged();
+      currentInfoChanged();
```
Comment (shoogle) [inline mscore/scoreaccessibility.cpp:355]: The `accessibleMessage` string is [cleared](https://github.com/musescore/MuseScore/blob/a3e23d6e63c3feff1058e6cf3616586785318af5/mscore/scoreaccessibility.cpp#L98) in the function `AccessibleScoreView::text()`, but that function only gets called if there is a screen reader running. If there is no screen reader then `accessibleMessage` would never be emptied after it was first set, hence `currentInfoChanged()` was never called to set the status bar text.

**Comment by MarcSabatella** [comment]
I can confirm it fixed this issue on Linux; can check on Windows with NVDA and JAWS on Friday to be sure I don’t see any other effects.

**Comment by shoogle** [comment]
@MarcSabatella, the thing to listen for is whether your `accessibleInfo` string is over-optimized, because now there will be occasions when `currentInfoChanged()` is called but the calculated `accessibleInfo` string will not get used. Next time `currentInfoChanged()` is called the string will get optimized again, possibly removing bar and beat numbers that should have been spoken.

**Comment by MarcSabatella** [comment]
Understood.  Most likely I won't be able to spend much time on this today.  It was all kind of a delicate balance before that mostly just worked by accidental of how NVDA happened to respond, but it would be good to figure out what's going on with JAWS too.  I should be able to find time for that next week.

BTW, so far making the timer thing not active on Linux seems an improvement overall, but I need to spend more time with that too.

File: mscore/scoreaccessibility.cpp [hunk 2]
```diff
@@ -93,10 +93,13 @@ QString AccessibleScoreView::text(QAccessible::Text t) const
...
                   QString msg = s->score()->accessibleMessage();
+                  QString info = s->score()->accessibleInfo();
                   if (msg.isEmpty())
-                        return s->score()->accessibleInfo();
+                        return info;
                   s->score()->setAccessibleMessage(""); // clear the message
-                  return msg;
+                  if (info.isEmpty())
+                        return msg;
+                  return tr("%1, %2").arg(msg).arg(info);
```
Comment (MarcSabatella) [inline mscore/scoreaccessibility.cpp:102]: Can you summarize when you expect this to hold?  Is it to read the mode text and note info?

Comment (shoogle) [reply mscore/scoreaccessibility.cpp:102]: Exactly. Typical values are:
```c++
msg  = "Steptime note input mode";
info = "Note C4 (quarter), beat 3, measure 1, staff 2";
```
`msg` could also relate to an action, such as `"Staccato above added"`, but `info` always describes the current selection.

**Comment by anatoly-os** [comment]
LGTM. @MarcSabatella do you confirm?

**Comment by MarcSabatella** [comment]
It does, although I haven't yet tested the lastest change to concatenate message & info.  At least I don't think that was in when I did my last test.  Will do so today, and do another round of checks on JAWS and Orca.

File: mscore/scoreaccessibility.cpp (hunk 2 above)
Comment (MarcSabatella) [reply mscore/scoreaccessibility.cpp:102]: Looking through the code paths, it does seem these would always be distinct - I had worried at first they might be the same.  Now the only thing I'd be concerned about is if there is ever a time when navigating through text (so msg is the character to read) that info would not have already been cleared and would still have the full text.  Not that I know of a case offhand, but that's what I'll still want to test.

**Comment by MarcSabatella** [comment]
Orca definitely does read the full text on each cursor movement, but it did that for the beta too, so this PR isn't a regression in that sense.  And things are smoother overall than the beta.  But maybe we should clear accessibleInfo when editing text?

Unfortunately, for reasons I don't yet understand, Orca is much slower to read than it was when I initially got it working just before FOSDEM.  But that doesn't seem related to these changes or to the previous changes made for beta - the alpha was slow in the same way.  Could be a result of other changes on my system, so I wouldn't worry about it.  Certainly nothing this PR is causing.

Will check Windows next.

**Comment by MarcSabatella** [comment]
Story is the same for NVDA and JAWS - they reads the full text of a text on each cursor movement (after reading the character at cursor).

BTW, some really good news here: JAWS is now reading the new messages for note input mode, adding/removing articulations, and navigation within text.  So to me, just clearing the info at the right point in text edit would be great.

File: mscore/scoreaccessibility.cpp (hunk 2 above)
Comment (shoogle) [reply mscore/scoreaccessibility.cpp:102]: @MarcSabatella, you were right to be concerned, hence I have explicitly emptied the info string during text editing. See [below](https://github.com/musescore/MuseScore/pull/6210/commits/628aa6f511ec5e647099e3c69083944ccb7cc675#diff-5f8c5742ec536a42250a6de753c09218R305).

**Comment by MarcSabatella** [comment]
Works great for my with Orca and NVDA.

With JAWS, reading of characters as I cursor through the text is erratic for some reason.  It wasn't working at all at first even with your changes, then it seemed it was for a while, now it is reading about half the time, and I can't see a pattern to it.  I wouldn't worry about it though, to me this is good to merge.

**Comment by shoogle** [comment]
@MarcSabatella, thanks for the review. @anatoly-os, should be good to go!

File: mscore/musescore.cpp [hunk 3]
```diff
@@ -4389,7 +4390,7 @@ void MuseScore::changeState(ScoreState val)
                   showPianoKeyboard(false);
                   break;
             case STATE_NORMAL:
-                  _modeText->hide();
+                  showModeText(tr("Normal mode"));;
```
Comment (Jojo-Schmitz) [inline mscore/musescore.cpp:4393]: This text doesn't show on Transifex, die to string freeze I guess
